### PR TITLE
feat: validate curated feed boundary

### LINKED ISSUE #5: [Task][P0][Backend] Failure-path tests and feed boundary validation
## Priority

`P0`

## Track

`backend`

## Summary

Add failure-path tests and stricter domain-boundary validation for the curated feed baseline.

## Context

`services/api` currently proves the happy path for `/health` and `/api/v1/feeds`, but it does not yet protect the failure path or validate the feed boundary as strictly as the next round of work needs.

This is the smallest backend hardening step needed before building more product logic on top of the scaffold.

## Scope

- write failing tests first for invalid or unsafe feed data at the boundary
- tighten validation around the curated feed descriptors
- keep route handlers thin
- preserve the existing success-path behavior unless validation requires an intentional error response
- keep the implementation focused on the baseline recovery only

## Planned Files

- `services/api/app/main.py`
- `services/api/app/services/feed_catalog.py`
- `services/api/tests/test_main.py`
- `services/api/tests/test_feed_catalog.py`

## Acceptance Criteria

- [ ] New tests cover at least one meaningful failure path
- [ ] Validation is enforced at the domain or API boundary instead of trusting raw data
- [ ] Existing success-path tests still pass
- [ ] `make test` and `make verify` pass

## Verification

- `make test`
- `make verify`

## Dependencies / Links

- Parent Epic: #4

### PR DESCRIPTION
**Pull request by dpwns523**
# 요약

- `/api/v1/feeds`의 curated feed 경계를 더 엄격하게 검증하도록 보강했습니다.
- 잘못된 feed 데이터가 API 경계에 들어오면 일관된 JSON 에러 응답을 반환하도록 처리했습니다.
- happy path뿐 아니라 failure path를 검증하는 테스트를 추가했습니다.

# 관련 이슈

- Closes #5

# 변경 사항

- `FeedDescriptor`에 non-empty 필드와 허용된 `kind` 검증 추가
- `FeedItem` 응답 모델에 stricter field validation 추가
- `/api/v1/feeds`에서 validation 예외를 `feed_validation_error` envelope로 변환
- invalid feed boundary를 검증하는 API 테스트 추가
- domain-level validation 테스트 추가

# 검증

- [x] `make bootstrap`
- [x] `make test`
- [x] `make verify`
- [x] 기타: API failure-path 테스트 추가 및 통과

# 영향 범위

- 사용자 영향: 잘못된 curated feed 데이터가 들어올 때 plain 500 대신 일관된 JSON 에러 응답을 받습니다.
- 운영 영향: curated feed 데이터 품질이 낮을 때 더 빨리 드러납니다.
- API/스키마 변경: `500` 에러 시 `{ "error": { "code", "message" } }` 응답이 명시됩니다.
- 배포 시 유의사항: 없음

# 참고 사항

- 이번 PR은 issue `#5` 범위만 다루며, 프론트 API 연동 이슈 `#6`은 포함하지 않았습니다.

### CODE REVIEW

File: services/api/app/main.py [hunk 1]
```diff
@@ -24,7 +36,22 @@ def read_health() -> dict[str, dict[str, str]]:
...
 
 
-@app.get("/api/v1/feeds", response_model=FeedResponse)
-def read_feeds() -> FeedResponse:
+def build_feed_response() -> FeedResponse:
     feeds = [FeedItem.model_validate(feed.__dict__) for feed in list_curated_feeds()]
     return FeedResponse(data=feeds, meta={"total": len(feeds)})
+
+
+@app.get("/api/v1/feeds", response_model=FeedResponse, responses={500: {"model": ErrorResponse}})
```
Comment (dpwns523) [inline services/api/app/main.py]: FastAPI 디렉토리 구조를 확장가능성이 있는 구조로 변경해야한다.

앞으로 개발할 API가 많으므로 main.py에서 관리하기 어려움

다음과 같은 구조를 추천
project/
├── main.py                  # FastAPI 애플리케이션 시작점
├── core/                    # 설정, 보안, 공통 유틸
│   ├── config.py            # 환경 변수 및 전역 설정
│   └── security.py          # 인증 및 JWT 처리
├── db/                      # 데이터베이스 설정
│   ├── base.py              # SQLAlchemy Base 선언
│   ├── session.py           # DB 엔진 및 세션 관리
│   └── migrations/          # Alembic 마이그레이션
├── models/                  # SQLAlchemy ORM 모델
│   ├── user.py
│   ├── item.py
│   └── ...
├── schemas/                 # Pydantic 요청/응답 스키마
│   ├── user.py
│   ├── item.py
│   └── ...
├── crud/                    # DB CRUD 처리 로직
│   ├── user.py
│   ├── item.py
│   └── ...
├── api/
│   └── v1/                  # API 버전별 구성
│       ├── endpoints/       # 실제 엔드포인트 정의
│       │   ├── user.py
│       │   ├── item.py
│       │   └── ...
│       └── routers.py       # 라우터 통합 및 등록
├── tests/                   # 테스트 코드
│   ├── test_user.py
│   ├── test_item.py
│   └── ...
└── celery_app.py            # Celery 설정 및 초기화


참고
- [https://github.com/fastapi/full-stack-fastapi-template]( https://github.com/fastapi/full-stack-fastapi-template)
- [https://github.com/zhanymkanov/fastapi-best-practices](https://github.com/zhanymkanov/fastapi-best-practices)

**Comment by dpwns523** [comment]
요청하신 FastAPI 디렉토리 구조 변경을 반영했습니다.

변경 이유
- 엔드포인트가 늘어날수록 `main.py` 단일 파일 구조는 라우트 등록, 응답 스키마, 경계 검증, 서비스 호출이 한 곳에 섞여 관리 비용이 커집니다.
- 그래서 앱 조립, 라우터 버전 구성, 엔드포인트, 스키마, 서비스 로직을 분리하는 방향으로 재구성했습니다.
- 코멘트에서 제안하신 전체 예시 구조를 한 번에 과도하게 스캐폴딩하기보다, 현재 PR 범위에 필요한 `api/v1/endpoints` + `schemas` + app factory 조합부터 적용해 이후 `core`, `db`, `models`, `crud`를 자연스럽게 확장할 수 있게 했습니다.

적용한 구조
- `services/api/app/main.py`: FastAPI app factory와 최상위 조립만 담당
- `services/api/app/api/router.py`: 최상위 router registry
- `services/api/app/api/endpoints/health.py`: 비버전 운영성 엔드포인트
- `services/api/app/api/v1/router.py`: v1 라우터 조합
- `services/api/app/api/v1/endpoints/feeds.py`: `/api/v1/feeds` 엔드포인트 정의
- `services/api/app/schemas/common.py`, `services/api/app/schemas/feed.py`: API boundary schema 분리
- `services/api/app/services/feed_catalog.py`: 기존 도메인/서비스 로직 유지

추가로 반영된 점
- 기존 경로(`/health`, `/api/v1/feeds`)는 그대로 유지했습니다.
- 이전에 넣었던 curated feed validation hardening도 새 구조로 옮겨 유지했습니다.
- `create_app()` 기준 라우트 등록 테스트를 추가했고, `make verify`까지 통과했습니다.

이제 다음 API가 추가되면 `app/api/v1/endpoints/*.py`와 `app/schemas/*.py`를 늘리고 `app/api/v1/router.py`에 연결하는 방식으로 확장할 수 있습니다.

**Comment by dpwns523** [comment]
FastAPI structure review 내용을 현재 브랜치에 맞춰 추가로 반영했습니다.

- docs/architecture.md에 현재 services/api 구조 검토 추가
- fastapi/full-stack-fastapi-template 및 fastapi-best-practices와의 차이와 정렬 방향 문서화
- 점진적인 목표 구조와 Mermaid 다이어그램 추가
- make verify 재실행 통과

이번 PR은 API 경계 검증 코드뿐 아니라, 그 구조가 왜 이렇게 시작되었는지와 다음 분리 방향까지 같이 리뷰할 수 있는 상태입니다.
